src/Model/Behavior/UploadBehavior.php

```diff
@@ -87,7 +87,7 @@ public function beforeSave(Event $event, Entity $entity, ArrayObject $options)
                 continue;
             }
 
-            if (Hash::get((array)$entity->get($field), 'error') !== UPLOAD_ERR_OK) {
+            if ((int)Hash::get((array)$entity->get($field), 'error') !== UPLOAD_ERR_OK) {
```

Comment:
Couldn't such a cast be problematic in case `Hash::get()` returns an empty-ish value like `null`, `false`, `''`, etc, as it would cast to `0`, wich equals `UPLOAD_ERR_OK`?

Reply:
@ndm2 Indeed.

@challgren This is not a safe change. You should debug and find out why the value of `error` is string for you instead of integer.

Reply:
So if i use `Cake\Controller\Controller` it returns an int. I'm thinking its related to https://github.com/dereuromark/cakephp-tools. Still doing debugging to figure out when and where its getting changed to a string.

Reply:
Ok so if I comment out `$this->loadComponent('Tools.Common');` it still is an int.

Reply:
So I can confirm this is because of https://github.com/dereuromark/cakephp-tools Common Component